### ISSUE TITLE
Add assigned-by oriented products to /reports

### ISSUE THREAD

**Issue by kltm** [geneontology/go-site#1944]
In discussion with @mugitty and @dustine32, using discussion from the managers' call and software call, we are outlining the "minimal" version of adding assigned-by products, without changing or refactoring current products.

The steps here would likely be:

- [x] use `combined.reports.json`  to create a new data object `assigned-by-combined-reports.json`, bascially a version that has been "pivoted" to be assigned-by oriented instead of file oriented
- [x] feed this new product through the same tools as the current report system to generate parallel "assigned-by-*" products 
- [ ] add this to the pipeline and expose these new products to the public
- [ ] get this information to the curating groups

**Comment by kltm** [comment]
@pgaudet @vanaukenk The updated data with the full load should be out now.

**Comment by mugitty** [comment]
From @pgaudet, ensure the following groups are also included in the report:
Alzheimers_University_of_Toronto
ASAP
AspGD
DFLAT
DIBU
GO_Noctua
GR
HPA
JaponicusDB
JCVI
LIFEdb
MENGO
MTBBASE
PAMGO_GAT
PAMGO_MGG
PAMGO_VMD
Roslin_Institute
SynGO-UCL
YuBioLab


Also, ordering should be case-insensitive.

**Comment by mugitty** [comment]
You are right @kltm only entries in the 'assigned-by' column with violations will be considered.

**Comment by pgaudet** [comment]
Hi, 

Thanks for the latest changes @mugitty 


I think this is ready to be folded in the release products - maybe named 
"gorule-report-by_contributor" ?

**Comment by kltm** [comment]
@pgaudet The reports are already using the prefix "assigned-by". I'd prefer to have a standard naming here: either the final report is "assigned-by-gorule-report" or all the other files are changed to "by_contributor-*" and this file is "by_contributor-gorule-report".

**Comment by pgaudet** [comment]
OK, which ever you prefer 

'assigned by' is what we have in the GAF specs (http://geneontology.org/docs/go-annotation-file-gaf-format-2.2/), so you can keep that.

**Comment by kltm** [comment]
@mugitty The final file should be named "assigned-by-gorule-report" or whatever the closest analog is to keep it with the other new report alphabetically.

**Comment by mugitty** [comment]
@pgaudet , @kltm.  Currently, the file name for the final file is named 'assigned-by-gorule-report.html'.  No file name changes should be required.     
I ran the pipeline code with changes requested by @pgaudet (output available in http://skyhook.berkeleybop.org/issue-go-site-1944-assigned-by-reports/reports/assigned-by-gorule-report.html) with the following changes:

1. The GORULE definition has been added after the link to the GO rule in the Contents section(Example http://skyhook.berkeleybop.org/issue-go-site-1944-assigned-by-reports/reports/assigned-by-AgBase-report.html)
2. The columns of the final report file (http://skyhook.berkeleybop.org/issue-go-site-1944-assigned-by-reports/reports/assigned-by-gorule-report.html) have been sorted case-insensitively.  Note, since we are using the same script for generating gorule-report.html(http://skyhook.berkeleybop.org/issue-go-site-1944-assigned-by-reports/reports/gorule-report.html) an unintended consequence is that the columns of this file has also been sorted in a case-insensitive manner. @pgaudet, is this change acceptable? or we can use a separate script for generating case-insensitive sort for 'assigned-by-gorule-report.html only.

**Comment by suzialeksander** [comment]
@pgaudet  I'm looking at http://skyhook.berkeleybop.org/issue-go-site-1944-assigned-by-reports/reports/assigned-by-gorule-report.html and there seem to be issues with the linked URLs- the column headers don't match the group in the columns

Just one example, but the column should be for SGD but the gorule-0000020 goes to http://skyhook.berkeleybop.org/issue-go-site-1944-assigned-by-reports/reports/assigned-by-pombase-report.html#gorule-0000020

![Screenshot 2023-03-13 at 15 07 37](https://user-images.githubusercontent.com/13735608/224833854-b401b951-e8bc-4714-824f-0ba343b14bc4.png)

**Comment by mugitty** [comment]
@suzialeksander, something is up with the column headers.  I need to look into it.  For now, you can get the sgd violations from  http://skyhook.berkeleybop.org/issue-go-site-1944-assigned-by-reports/reports/assigned-by-sgd-report.html
Thanks for reporting

**Comment by kltm** [comment]
@mugitty From the annotation call this morning, there was a question about where the `.md` versions of the report files were? I'm not sure if they are being produced, but are not copied over, if they are not being produced, or if there is something else we need to look into. I wanted to check in with you first.

**Comment by mugitty** [comment]
@kltm, a .md version of the report file is not being produced. Only an assigned-by-*.html file for groups with GORULE violations is being produced. There was no equirement to produce a .md file

**Comment by kltm** [comment]
@mugitty No worries--thank you for the clarification.
@pgaudet Is the intent here to expand the scope for this?

**Comment by kltm** [comment]
@dustine32 Okay, adding here as it took me a little to figure out: the `*-reports.md` files are actually created in the pipeline mega-step; similar to some reprocessing that happens in the Makefile-gaf-reprocess. Does that sound correct?
@mugitty @pgaudet With that, we can't really make an md version of the assigned-by reports without some hacking in there or just re-creating them wholesale from the JSON that we have. Frankly, I'd vote the latter, as it's more flexible to work from a "database" than to continue to make the mega-make even more full of quirks when we're looking to replace it. @pgaudet I don't think it would be too hard, but that would be your call for the scope change.

**Comment by dustine32** [comment]
Right @kltm, pipeline mega-step generate `.md` reports in both [validate.py](https://github.com/biolink/ontobio/blob/8d5defc805d0d48017b149e5c6fde6ebb6f9f845/bin/validate.py#L256) and [ontobio-parse-assocs.py](https://github.com/biolink/ontobio/blob/8d5defc805d0d48017b149e5c6fde6ebb6f9f845/bin/ontobio-parse-assocs.py#L187) by calling `to_markdown()` on the Report object of `GafParser` (or `GpadParser`).

**Comment by mugitty** [comment]
I would prefer creating the .md file from the assigned-by combined report json file.

**Comment by kltm** [comment]
@mugitty I think that you are right about the approach to making a .md file. 
It's worth noting that what's useful to users in the .md is that they can copy and paste directly into a spreadsheet, so that is the main thing to preserve in the .md recreation.
Honestly though, if that it the only thing they want, it may be easier to just change the current HTML output to better preserve tabs and spaces in the HTML generation.
It would be good to get feedback from @pgaudet before committing.

**Comment by kltm** [comment]
@mugitty With what @pgaudet has said, I believe the easiest next steps would be to: 
1.  improve the html output so that copy-and-paste to spread works as it does in the .md
2. if that doesn't seem to be working for some reason, recreate the .md version as closely as possible using the JSON blob with a new script (I'd also like to see that new script take over, but this is the fallback case so let's leave that there for now)

This would be better spun out as a new ticket for this project; @pgaudet the scope of this issue is complete with a little more testing.

**Comment by mugitty** [comment]
> 

@suzialeksander, the table in http://skyhook.berkeleybop.org/issue-go-site-1944-assigned-by-reports/reports/assigned-by-gorule-report.html is now sorted properly.  Thanks for reporting

**Comment by kltm** [comment]
Reopening: we still need to get this cleared through the main pipelines and get the information to people in the docs (as above).

**Comment by kltm** [comment]
@pgaudet This is now cleared and can be closed once "sufficient communication" with curators has occurred.

**Comment by kltm** [comment]
Closing this issue closes the project.
